cardano-crypto-class/src/Cardano/Crypto/DSIGN/EcdsaSecp256k1.hs

```diff
@@ -55,56 +80,184 @@ import Cardano.Crypto.DSIGN.Class (
+            res <- secpEcPubkeyCreate ctx vkp skp
+            when (res /= 1) 
+                 (error "deriveVerKeyDSIGN: Failed to derive VerKeyDSIGN EcdsaSecp256k1DSIGN")
+    {-# NOINLINE signDSIGN #-}
+    signDSIGN () (MH psb) (SignKeyEcdsaSecp256k1 skBytes) = 
+      SigEcdsaSecp256k1 <$> unsafeDupablePerformIO . psbUseAsSizedPtr psb $ \psp -> do
+        psbUseAsSizedPtr skBytes $ \skp ->
+          psbCreateSized $ \sigp -> 
+            withForeignPtr secpCtxPtr $ \ctx -> do
+              -- The two nullPtr arguments correspond to nonces and extra nonce
+              -- data. We use neither, so we pass nullPtrs to indicate this to the
+              -- C API.
+              res <- secpEcdsaSign ctx sigp psp skp nullPtr nullPtr
+              when (res /= 1) 
+                   (error "signDSIGN: Failed to sign EcdsaSecp256k1DSIGN message")
+    {-# NOINLINE verifyDSIGN #-}
+    verifyDSIGN () (VerKeyEcdsaSecp256k1 vkBytes) (MH psb) (SigEcdsaSecp256k1 sigBytes) = 
+      unsafeDupablePerformIO . psbUseAsSizedPtr psb $ \psp -> do
+        psbUseAsSizedPtr sigBytes $ \sigp -> 
+          psbUseAsSizedPtr vkBytes $ \vkp -> 
+            withForeignPtr secpCtxPtr $ \ctx -> do
+              let res = secpEcdsaVerify ctx sigp psp vkp
+              pure $ case res of 
+                0 -> Left "verifyDSIGN: Incorrect or unparseable SigDSIGN EcdsaSecp256k1DSIGN"
+                _ -> Right ()
+    genKeyDSIGN seed = runMonadRandomWithSeed seed $ do
+      bs <- getRandomBytes 32
+      case psbFromByteStringCheck bs of 
+        Nothing -> error "genKeyDSIGN: Failed to generate SignKeyDSIGN EcdsaSecp256k1DSIGN unexpectedly"
+        Just psb -> pure $ SignKeyEcdsaSecp256k1 psb
+    {-# NOINLINE rawSerialiseSigDSIGN #-}
+    rawSerialiseSigDSIGN (SigEcdsaSecp256k1 psb) = 
+      psbToByteString @SECP256K1_ECDSA_SIGNATURE_BYTES . unsafeDupablePerformIO $ do
+        psbUseAsSizedPtr psb $ \psp -> 
+          psbCreateSized $ \dstp ->
+            withForeignPtr secpCtxPtr $ \ctx -> 
+              void $ secpEcdsaSignatureSerializeCompact ctx dstp psp
+    {-# NOINLINE rawSerialiseVerKeyDSIGN #-}
+    rawSerialiseVerKeyDSIGN (VerKeyEcdsaSecp256k1 psb) = 
+      psbToByteString . unsafeDupablePerformIO . psbUseAsSizedPtr psb $ \psp ->
+        psbCreateLen @SECP256K1_ECDSA_PUBKEY_BYTES $ \ptr len -> do
+          let dstp = castPtr ptr
+          -- This is necessary because of how the C API handles checking writes:
+          -- maximum permissible length is given as a pointer, which is
+          -- overwritten to indicate the number of bytes we actually wrote; if
+          -- we get a mismatch, then the serialization failed. While an odd
+          -- choice, we have to go with it.
+          alloca $ \(lenPtr :: Ptr CSize) -> do
+            poke lenPtr len
+            withForeignPtr secpCtxPtr $ \ctx -> do
+              void $ secpEcPubkeySerialize ctx dstp lenPtr psp secpEcCompressed
```

Comment:
I don't think using `void` is a good idea without pinning it's `f a` type to something, or just consume the return value in a trivial way. If `secpEcPubkeySerialize` ever change signature (e.g. add parameters / return value), this would not be caught by the type system.

Reply:
This strikes me as incredibly unlikely: the only way this could happen is if `secp256k1` introduced a breaking change in the function this wraps, _and_ we upgrade our Nix setup in such a way as to introduce the library version with this breaking a change. 

If our goal is to future-proof against such a thing, the work must go _well_ beyond what you are suggesting. I am happy to consider this, but only if this would be done in a systematic way across the entire project: there is no point future-proofing one small call when the rest of the library doesn't receive similar treatment.

@lehins - thoughts?

Reply:
I am actually agreeing with @vincenthz on this. It will be good to future proof this function, because there is nothing besides the secp56k1 library's documentation that enforces that the output is always `1`. All it takes is a change to `secp256k1_ec_pubkey_serialize` implementation and we are no longer guarded against errors, because I am more than sure that either no one will notice documentation change and/or will remember to update this call site

I'v made this change here: https://github.com/input-output-hk/cardano-base/pull/296

_side note_ - @kozross it was invisible in this PR's diff, but once I opened source files in my editor, I noticed a whole lot of trailing whitespace. I highly recommend to adjust your editor to remove them on save.